### PR TITLE
OG-173 relaydc deployment test

### PR DESCRIPTION
**Pull request by drortirosh**
Full deployment test:
`./dockers/test/testMultiRelayer/test.sh`

- bring up a node (ganache) 
- gsn-deploy (including VersionRegistry, and tagging RelayHub)
- create custom (test) server with 2 relayer instances
- register-relayer
- validate both instances are active (and different..)

modules:
- `dockers/relaydc` - the "root" server docker image to run. see `readme`. 
   `rdc` - launcher script
- `dockers/cli` - docker wrapper for `gsn` command lines (after build, launch with `./gsn`)
    supports: deploy, paymaster, relay-register, registry
    **removed**: start, relayer-run

### CODE REVIEW

File: dockers/relaydc/docker-compose.yml [hunk 1]
```diff
@@ -50,6 +50,8 @@ services:
       port: 80
       workdir: /app/data
       config: /app/config/gsn-relay-config.json
+      relayHubId: hub
```
Comment (shahafn) [inline dockers/relaydc/docker-compose.yml:53]: Is it address? maybe rename to relayHubAddress?

File: dockers/tests/testMultiRelayer/r [hunk 2]
```diff
@@ -0,0 +1,2 @@
+docker run --rm -ti -v /var/run/docker.sock:/var/run/docker.sock -e ROOT=$PWD -v $PWD/.env:/.env opengsn/relaydc:test2 "$@"
```
Comment (shahafn) [inline dockers/tests/testMultiRelayer/r:1]: Please rename 'r' to something that is more than one letter.

File: src/cli/commands/gsn.ts [hunk 3]
```diff
@@ -1,14 +1,16 @@
 #!/usr/bin/env node
 
 import commander from 'commander'
+import { version } from '../../../package.json'
```
Comment (shahafn) [inline src/cli/commands/gsn.ts:4]: I remember that this was an issue before...

File: src/cli/utils.ts [hunk 4]
```diff
@@ -80,23 +86,23 @@ export function showDeployment (deploymentResult: DeploymentResult, title: strin
...
   Forwarder: ${deploymentResult.forwarderAddress}
   Paymaster ${paymasterTitle != null ? '(' + paymasterTitle + ')' : ''}: ${deploymentResult.naivePaymasterAddress}`)
 }
 
 export function loadDeployment (workdir: string): DeploymentResult {
   function getAddress (name: string): string {
-    const address = getAddressFromFile(path.join(workdir, name + '.json'))
-    if (address != null) { return address }
-    throw new Error('no address for ' + name)
+    return getAddressFromFile(path.join(workdir, name + '.json')) as string
```
Comment (shahafn) [inline src/cli/utils.ts:96]: Why did you remove the error check?

File: src/common/VersionRegistry.ts [hunk 5]
```diff
@@ -78,4 +100,30 @@ export class VersionRegistry {
       })
       .reverse()
   }
+
+  // return all IDs registered
+  async listIds (): Promise<string[]> {
+    const events = await this.registryContract.getPastEvents('VersionAdded', { fromBlock: 1 })
```
Comment (shahafn) [inline src/common/VersionRegistry.ts:106]: Can't you use ContractInteractor to wrap calls with web3?

File: dockers/cli/readme [hunk 6]
```diff
@@ -0,0 +1,7 @@
+docker image wrapper for the gsn command-line tool
+
+use the `gsn` wrapper script to launch.
+NOTE: only current folder is mapped into the docker, so if you reference external file,
+make sure it is under the current directory
```
Comment (shahafn) [inline dockers/cli/readme:5]: "."

File: src/cli/commands/gsn.ts (hunk 3 above)
Comment (drortirosh) [reply src/cli/commands/gsn.ts:4]: It works with `npx gsn -V` from another project, as well as with `tsnode ./src/cli/commands/gsn.ts -V`  and with `node ./dist/cli/commands/gsn.js`
And of course, it works with the docker wrapper by running `dockers/gsn/gsn -V`

So I think its OK...

File: dockers/tests/testMultiRelayer/r (hunk 2 above)
Comment (drortirosh) [reply dockers/tests/testMultiRelayer/r]: done. its now "testrdc", as its test-only wrapper for rdc

**Review by shahafn — APPROVED** [review]
